include/itkCuberilleImageToMeshFilter.hxx

```diff
@@ -450,6 +450,10 @@ CuberilleImageToMeshFilter<TInputImage,TOutputMesh,TInterpolator>
     {
     // Compute normal vector
     normal = m_GradientInterpolator->Evaluate( vertex );
+    if (normal.GetSquaredNorm() == 0)
```

Comment:
This problem (gradient==0) appears when the input image is a binary mask. If norm is non-zero (even 1e-96) that will lead to a non-NaN vector which is fine.